src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -355,14 +355,46 @@ static void setCondsAndInitialNotify
       status = subOrig.hasField(CSUB_STATUS)? getStringFieldF(subOrig, CSUB_STATUS) : STATUS_ACTIVE;
     }
 
-    std::string url;
+    HttpInfo httpInfo;
     if (subUp.notificationProvided)
     {
-      url = subUp.notification.httpInfo.url;
+      httpInfo = subUp.notification.httpInfo;
```

Comment:
As far as I see it, you'd just need an extra 'if' to avoid this copy.
[ Or even better, use a pointer ]

Never mind. I am just a bit tired of unnecessary copies of objects ...

NTC